src/test/java/org/mvel2/tests/core/ASMConsistencyTest.java

```diff
@@ -41,5 +41,31 @@ public void run() {
       throw new IllegalStateException("Exception occurred " + rex.size() + " time(s)", rex.get(0));
     }
   }
+  
+  /**
+   * used for asm optimize test
+   */
+  public static class TestFunction {
 
+    public static boolean isNull(String str) {
+      return str == null;
+    }
+
+  }
+
+  public void testNullArgConvert() {

```

Comment:
Hi @SRWFan , I ran this test without you code change and it succeeded so it may not test the actual change.

Reply:
Hi @MalcolmOdd, This bug depends on the expression evalute speed.So I changed DynamicOptimizer's props to make sure it always happens.You can try it again.

Reply:
@SRWFan thanks, I re-ran it with your change and was able to reproduce it